src/Mesh/Interpolation.jl

```diff
@@ -0,0 +1,454 @@
+        x2g = range(xbnd[1,2], xbnd[2,2], step=xres[2])
+        x3g = range(xbnd[1,3], xbnd[2,3], step=xres[3]) 
+        #-----------------------------------------------------------------------------------
+        realelems =  grid.topology.realelems # Element (numbers) on the local processor
+        Nel       = length(realelems)
+        n123      = zeros(T,    ndim)     # # of unique ξ1, ξ2, ξ3 points in each cell
+        xsten     = zeros(T, 2, ndim)     # x1, x2, x3 start and end for each brick element
+        xbndl     = zeros(T, 2, ndim)     # location of x1,x2,x3 limits (min,max) for each brick element
+
+        ξ1 = map( i -> zeros(FT,i), zeros(T,Nel))
+        ξ2 = map( i -> zeros(FT,i), zeros(T,Nel))
+        ξ3 = map( i -> zeros(FT,i), zeros(T,Nel))
+        V  = map( i -> zeros(FT,i), zeros(T,Nel))
+        x  = map( i -> zeros(FT,ndim,i), zeros(T,Nel)) # interpolation grid points embedded in each cell 
+        #-----------------------------------------------------------------------------------
+        for el in 1:Nel
+            for (ξ,xg,dim) in zip((ξ1,ξ2,ξ3), (x1g, x2g, x3g), 1:ndim)
+                xbndl[1,dim], xbndl[2,dim] = extrema(grid.topology.elemtocoord[dim,:,el])
+                xsten[1,dim], xsten[2,dim] = findfirst( temp -> temp ≥ xbndl[1,dim], xg ), findlast( temp -> temp ≤ xbndl[2,dim], xg )
+                n123[dim] = xsten[2,dim] - xsten[1,dim]
+                ξ[el] = [ 2 * ( xg[ xsten[1,dim] + i - 1] - xbndl[1,dim] ) / (xbndl[2,dim]-xbndl[1,dim]) -  1 for i in 1:n123[dim] ]
+            end
+
+            x_el  = zeros(FT,ndim,prod(n123))
+            V[el] = zeros(FT,prod(n123))
+
+            ctr = 1
+
+            for k in 1:n123[3], j in 1:n123[2], i in 1:n123[1]
+                x_el[1,ctr] = x1g[ xsten[1,1] + i - 1 ]
+                x_el[2,ctr] = x2g[ xsten[1,2] + j - 1 ]
+                x_el[3,ctr] = x3g[ xsten[1,3] + k - 1 ]
+                ctr += 1
+            end
+            x[el] = x_el
+        end # el loop
+        #-----------------------------------------------------------------------------------
+        return new{FT}(realelems, xbnd, xres, ξ1, ξ2, ξ3, x, V)
+    end
+#--------------------------------------------------------
+end # struct InterpolationBrick 
+#--------------------------------------------------------
+"""
+    interpolate_brick!(intrp_brck::InterpolationBrick, sv::AbstractArray{FT}, st_idx::T, poly_order::T) where {T <: Integer, FT <: AbstractFloat}
+
+This interpolation function works for a brick, where stretching/compression happens only along the x, y & z axis.
+Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
+
+# input
+ - `intrp_brck` Initialized InterpolationBrick structure
+ - `sv` State vector
+ - `st_idx` # of state vector variable to be interpolated
+ - `poly_order` polynomial order for the simulation
+"""
+function interpolate_brick!(intrp_brck::InterpolationBrick, sv::AbstractArray{FT}, st_idx::T, poly_order::T) where {T <: Integer, FT <: AbstractFloat}
```

Comment:
we should make `poly_order` a field of the `InterpolationBrick` struct, since it is defined by the `grid`.